CMakeLists.txt

```diff
@@ -1,142 +1,108 @@
-# DAGMC Exporter
-###############################################################################
-if(${BUILD_DAGMC_EXPORTER})
-    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBUILD_DAGMC_EXPORT" )
-
-    find_package(MOAB REQUIRED)
-    include_directories(${MOAB_INCLUDE_DIRS})
-
-    # Find the other libraries and directories
-    include_directories("${DAGMC_DIR}/include")
-
-    find_library(MAKE_WATERTIGHT NAMES makeWatertight PATHS "${DAGMC_DIR}/lib" )
-
-    set(DAGMC_SRC
-        export_dagmc_cmd/DAGMCExportCommand.cpp
-        export_dagmc_cmd/DAGMCExportCommand.hpp)
-
-      
-endif()
-
-###############################################################################
-# MCNP Importer
-###############################################################################
-if(${BUILD_MCNP_IMPORTER})
-    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBUILD_MCNP_IMPORT" )
-    set( BUILD_IGEOM ON)
-
-    include_directories("${CMAKE_SOURCE_DIR}/iGeom")
-
-    add_subdirectory(mcnp2cad)
-    include_directories(mcnp2cad)
-
-    set( MCNP2CAD_LIB "mcnp2cad")
-    set(MCNP_SRC
-        import_mcnp_cmd/MCNPImp.cpp
-        import_mcnp_cmd/MCNPImp.hpp)
-
-
-endif()
-
-
-###############################################################################
-# iGeom Tests
-###############################################################################
-
-if(${BUILD_IGEOM_TESTS})
-  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBUILD_IGEOM_TESTS" )
-  set( BUILD_IGEOM ON)
-
-    include_directories("${CMAKE_SOURCE_DIR}/iGeom")
-
-    set(IGEOMTEST_SRC
-      iGeom/tests/iGeom_test.cpp
-      iGeom/tests/iGeom_test.hpp)
-
-endif()
-
-###############################################################################
 # iGeom
-###############################################################################
-
-
-if(${BUILD_IGEOM})
-  SET(IGEOM_SRC
-      iGeom/iGeom.cpp
-      iGeom/iGeom.h
-      iGeom/iGeomError.cc)
-
-  add_library(iGeom SHARED ${IGEOM_SRC})
-  target_link_libraries(iGeom cubit_geom ${CUBITI19})
-
-
-
-endif()
-
-###############################################################################
-# Build commands
-###############################################################################
-
-add_library(svalinn_plugin MODULE ${PLUGIN_SRC} ${DAGMC_SRC} ${MCNP_SRC} ${IGEOMTEST_SRC})
-target_link_libraries(svalinn_plugin cubiti cubit_util cubit_geom ${MAKE_WATERTIGHT} ${MOAB_LIBRARIES} ${MCNP2CAD_LIB})
-
-if(${BUILD_IGEOM})
-    add_dependencies(svalinn_plugin iGeom)
-endif()
-
-if(${BUILD_MCNP_IMPORTER})
-  add_dependencies(svalinn_plugin mcnp2cad)
-  add_dependencies(mcnp2cad iGeom)
-endif()
+if (BUILD_IGEOM)
+  message(STATUS "Building iGeom")
+  set(IGEOM_SRC_FILES iGeom/iGeom.cpp iGeom/iGeomError.cc)
+  include_directories(iGeom)
+  add_library(iGeom SHARED ${IGEOM_SRC_FILES})
+  target_link_libraries(iGeom ${CUBIT_LIBS})
+  install(TARGETS iGeom LIBRARY DESTINATION lib)
+  list(APPEND PLUGIN_LINK_LIBS iGeom)
+  set(IGEOM_LIB iGeom)
```

Comment:
I don't know how else to do it. If we really want to use an `IGEOM_LIB` variable, then I don't know how else to do it

Reply:
Maybe left our conversation today with the wrong idea. I thought we ended up deciding that we'd set the `STANDALONE_MCNP2CAD` variable here instead of relying on `IGEOM_LIB`

Reply:
But in [this](https://github.com/svalinn/mcnp2cad/pull/68#discussion_r483749685) comment, you talk about wanting to use IGEOM_LIB. I'm confused about what it is we're trying to do

Reply:
Ah, I see. In that comment I meant to say that we should rely on a CMake target for iGeom rather than calling `link_directories` and `include_directories`, which could cause problems for other targets in the future. My use of `IGEOM_LIB` as a name for that target was confusing though. Apologies.

Reply:
My understanding is that we want to set this `IGEOM_LIB` variable so that is can be used in linking the `mcnp2cad` library.  Perhaps the debate is about how we set it.  @pshriwise had used the line here: https://github.com/svalinn/Trelis-plugin/pull/59/files#diff-af3b638bc2a3e6c650974192a53c7291R89 which seems like it perhaps accomplishes the same thing, but brings some CMAKE semantics along for the ride.

Reply:
Right. Just to lay out how I understand the path forward based on our video call and subsequent messages in these PRs:

We need to be able to indicate to the `mcnp2cad` project that it's being built as part of the plugin and doesn't need to search for `iGeom` because it will be provided as part of the plugin build. We decided to do that by setting the `STANDALONE_MCNP2CAD` option to `OFF` in the plugin before calling `add_subdirectory(mcnp2cad)`.

Somewhat separate from this is how we link to `iGeom` in the `mcnp2cad` project. Right now in `mcnp2cad` we find the include and library paths, and then add make update the include and link directories of the CMake project, but I'd prefer that we rely on creating a CMake target to link to when creating the `mcnp2cad` library/executable. The motivation for this is that the additional information in these project variables for the includes and link dirs can pollute other targets with unused flags and in the worst case can cause errors, which may be hard to debug down the road.

(:point_up: Somewhat separate...) If we are relying on a target for `iGeom` in `mcnp2cad`, then we'll want to provide the `iGeom` target we create in the plugin to `mcnp2cad` rather than redefining information we already have in the subproject, hence the (poorly named) alias I was creating in #59 to match the name of what I thought was a target in `mcnp2cad`, `IGEOM_LIB`. (I unfortunately misunderstood that our `find_library` call is really defining a variable not a target, but due to the syntax used in `mcnp2cad` when linking to `iGeom` this kind of worked out for me.)

Please correct me if I'm off point on something above, but if not here's what I see as a valid path forward for these PR's:

 In `mcnp2cad`:

 - when `STANDALONE_MCNP2CAD` is `ON`, use information from the `find_library` and `find_path` calls in `mcnp2cad` to create an imported target for `iGeom` (an example this for a DAGMC target is [here](https://github.com/openmc-dev/openmc/pull/1499/files))

In `trelis-plugin`:

  - set `STANDALONE_MCNP2CAD` accordingly
  - create a target alias for the `iGeom` target created in the plugin that matches the name of the `iGeom` target in `mcnp2cad`